src/wp-includes/class-wp-query.php

```diff
@@ -3221,7 +3221,7 @@ public function get_posts() {
 		}
 
 		if ( null === $this->posts ) {
-			$split_the_query = ( $old_request == $this->request && "{$wpdb->posts}.*" === $fields && ! empty( $limits ) && $q['posts_per_page'] < 500 );
+			$split_the_query = ( $old_request == $this->request && "{$wpdb->posts}.*" === $fields && ( wp_using_ext_object_cache() || ( ! empty( $limits ) && $q['posts_per_page'] < 500 ) ) );
```

Comment:
```suggestion
			$split_the_query = (
				$old_request == $this->request
				&& "{$wpdb->posts}.*" === $fields
				&& (
					wp_using_ext_object_cache()
					|| ( ! empty( $limits ) && $q['posts_per_page'] < 500 )
				)
			);
```
This conditional is super hard to read, let's split it into shorter lines 🙂